src/java/picard/analysis/artifacts/Transition.java

```diff
@@ -51,4 +46,46 @@ public Transition matchingRef() {
     public String toString() {
         return this.ref + ">" + this.call;
     }
+
+    protected enum Base {
+        A ('A'),
+        C ('C'),
+        G ('G'),
+        T ('T');
+
+        public byte base;
+
+        private Base(final char base) {
+            this.base = (byte)base;
+        }
+    }
+
+    static protected final int[] baseIndexMap = new int[256];
+    static {
+        Arrays.fill(baseIndexMap, -1);
+        baseIndexMap['A'] = Base.A.ordinal();
+        baseIndexMap['C'] = Base.C.ordinal();
+        baseIndexMap['G'] = Base.G.ordinal();
+        baseIndexMap['T'] = Base.T.ordinal();
+    }
+
+    static private final Transition[][] transitionIndexMap = new Transition[4][4];
+    static {
+        transitionIndexMap[0][0] = AtoA;
```

Comment:
valueOf doesn't work for enums with multiple values (that's why Matt had a String addition thing in the previous version).  This is the cleanest it's going to get.